### PR TITLE
Caherbel/high level wip

### PR DESCRIPTION
**Pull request by cherbel**
Creating Tests for AdafruitMAX31865 using Arduino Continuous Integration. 

-Low level tests for testing pin I/O.
-High level tests for functionality testing.

### CODE REVIEW

File: .arduino-ci.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+unittest:
+  platforms:
+    - mega2560
+  testfiles:
+    reject:
+      - "Common.cpp"
+
+compile:
+  platforms:
+    - mega2560
```
Comment (jgfoster) [inline .arduino-ci.yml]: Git tends to like a newline at the end of files.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+/spec/reports/
+vendor
+*.gem
+
+# rspec failure tracking
+.rspec_status
+
+# C++ stuff
+*.bin
+*.bin.dSYM
```
Comment (jgfoster) [inline .gitignore]: newline

**Comment by jgfoster** [comment]
Remind me why the Adafruit_SPIDevice.h and .cpp files were added.

**Comment by cherbel** [comment]
> Remind me why the Adafruit_SPIDevice.h and .cpp files were added.

@jgfoster Adafruit_SPIDevice.cpp and .h were added because the Adafruit_MAX31865 library is dependent on those files. I think that I/O is handled in Adafruit_SPIDevice if I remember correctly.

**Comment by cherbel** [comment]
@jgfoster I made the changes you requested above. Let me know if I can do anything else.

**Comment by jgfoster** [comment]
As discussed, let's see if we can get Adafruit_SPIDevice.h and .cpp as a prerequisite rather than having them duplicated.

**Comment by jgfoster** [comment]
Is there a reason this was closed? Are you going to submit a new PR later?

**Comment by cherbel** [comment]
> Is there a reason this was closed? Are you going to submit a new PR later?

This pull request was resolved in #2 . This pull request was made before I pulled out the SPI_Devices files.

**Pull request closed without merging**